### PR TITLE
add appnp for small dataset

### PR DESCRIPTION
**Pull request by DesmonDay**

### CODE REVIEW

File: examples/kddcup2021/WikiKG90M/post_smoothing/appnp.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    return new_entity_feat
+        
+def ppnp_ote(entity_feat_path, relation_feat_path, entity_neighbors, alpha, gamma=10):
+    new_entity_feat = np.zeros(entity_feat.shape, dtype="float32")
+    
+
+# For FB15k-237 and WN18RR.
+def main_small(model, dataset, alpha=0.98, k_hop=10, gamma=10):
+    data_path = os.path.join('/home/work/suweiyue/Shared/WikiKG90M/RPS/KGEmbedding-OTE/data/', dataset)
+    feat_path = os.path.join('/home/work/suweiyue/Shared/WikiKG90M/RPS/KGEmbedding-OTE/models', '%s_%s' % (model, dataset))
```
Comment (Yelrose) [inline examples/kddcup2021/WikiKG90M/post_smoothing/appnp.py:96]: The name of the directory should be anonymous.

File: examples/kddcup2021/WikiKG90M/post_smoothing/appnp.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        edges = np.array(edges, dtype=np.int32)
+        
+    # 构造邻居部分
+    n_entities = len(entity2id)
+    # 格式 一个实体的邻居：[[a, b, c], [r1, r2, r3], [1, -1, 1]](包括方向)，多个实体就用列表存起来
+    entity_neighbors = []
+    for eid in tqdm(range(n_entities)):
+        nb = [[], [], []]
+        # eid位于头实体
+        h_index = np.where(edges[:, 0] == eid)
```
Comment (Yelrose) [inline examples/kddcup2021/WikiKG90M/post_smoothing/appnp.py:38]: 复杂度略高

Comment (Yelrose) [reply examples/kddcup2021/WikiKG90M/post_smoothing/appnp.py:38]: O(N * E)

**Review by Yelrose — CHANGES_REQUESTED** [review]

File: examples/kddcup2021/WikiKG90M/post_smoothing/appnp.py [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+        for line in fin:
+            h, r, t = line.strip().split('\t')
+            hid = entity2id[h]
+            rid = relation2id[r]
+            tid = entity2id[t]
+            edges.append([hid, rid, tid])
+        edges = np.array(edges, dtype=np.int32)
+        
+    n_entities = len(entity2id)
+    # 格式 实体的邻居：[[a, b, c], [r1, r2, r3], [1, -1, 1]](包括方向)，多个实体就用列表存起来
```
Comment (Liwb5) [inline examples/kddcup2021/WikiKG90M/post_smoothing/appnp.py:36]: 尽量不要用中文注释. 特别是文件开头没有加上: #-*- coding: utf-8 -*-

**Pull request closed without merging**